.github/workflows/release-artifacts.yml

```diff
@@ -116,18 +122,29 @@ jobs:
       - name: Install cibuildwheel
         run: python -m pip install cibuildwheel==2.9.0 poetry==1.2.0
 
-      # Only build a single wheel in CI.
-      - name: Set env vars.
-        run: |
-          echo "CIBW_BUILD="cp37-manylinux_x86_64"" >> $GITHUB_ENV
+      - name: Set up QEMU to emulate aarch64
+        if: matrix.arch == 'aarch64'
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64
+
+      - name: Build aarch64 wheels
+        if: matrix.arch == 'aarch64'
+        run: echo 'CIBW_ARCHS_LINUX=aarch64' >> $GITHUB_ENV
+
+      - name: Only build a single wheel on PR
         if: startsWith(github.ref, 'refs/pull/')
+        run: echo "CIBW_BUILD="cp37-manylinux_${{ matrix.arch }}"" >> $GITHUB_ENV
```

Comment:
I think this would mean we're going to build `cp37-manylinux_x86_64` and `cp37-manylinux_aarch64` on PRs?

AFAIK we want to build arm wheels/debs/docker on release branches only (since qemu makes everything sluggish); though it's nice to have this here so that it proves the workflow works.

Reply:
Jep, I saw that your previous approach didn't have the aarch64 builds on PRs. I wanted to have it running at least once, to verify it works. I think it might be not bad in general, since aarch64 builds may not find pre-compiled wheels for dependencies, e. g. on dependency version changes, which we might want to catch before merging into `develop`? But of course I can exclude it on PR, if you prefer, or can do so at a later time, when it appears to be annoying on every PR sync.

Reply:
We would prefer to have arm excluded from PRs, please---for pragmatism's sake. (We have a limited number of GHA runners available and quite a lot of PRs to juggle!)

Reply:
For completeness: the "Wheels" archive on https://github.com/matrix-org/synapse/actions/runs/3273849733 contains an aarch64 and x86_64 wheel. Looks promising!

Reply:
`aarch64` on PR CI excluded now. I left the `${{ matrix.arch }}` in place here, making it easier to test in future PRs editing the `exclude` block only.

I have a NanoPi R4S just up and running and see whether I can get that wheel installed. Never installed a wheel manually, AFAIK, let's see how that works 😅.

Reply:
Tried it on my R5S without success

```
root@DietPiR5S:/tmp# pip install matrix_synapse-1.69.0-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
ERROR: matrix_synapse-1.69.0-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl is not a supported wheel on this platform.
root@DietPiR5S:/tmp#
```

Maybe I'm doing it wrong way

Reply:
It may be due to the Python 3.7 lower and upper version boundaries. I even have a Debian Bookworm system here with Python 3.10, I guess yours is Bullseye Python 3.9? I'll try with a Buster container.

Reply:
>  guess yours is Bullseye Python 3.9?
Yes, our default image on R5S

Reply:
It certainly seems to have the right architecture:

```
$ file /home/dmr/Wheel/matrix_synapse-1.69.0-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64/synapse/*.so
/home/dmr/Wheel/matrix_synapse-1.69.0-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64/synapse/synapse_rust.abi3.so: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, BuildID[sha1]=eab3446b7d17a7de08fefa3391f4c397df82d013, with debug_info, not stripped
```

But yeah, I think this will only be installable against a CPython 3.7 on aarch64.

Reply:
Brilliant---thank you for confirming.